lisp/ess-mode.el

```diff
@@ -429,32 +421,39 @@ current function."
 
 
 ;;; Formatting / indentation
+(defvar-local ess--installed-style-vars nil
+  "A cons of the form (STYLE . VARS).
+VARS is a list of all style vars which were not set explicitly to
+buffer local values by the user in mode hooks.")
 
-(defun ess-set-style (&optional style quiet)
+(defun ess-set-style (&optional style _quiet)
```

Comment:
Is style mandatory now? It should go before `&optional`, right?

Reply:
No, it's not. When it's nil, it means install `ess-style` but don't overwrite the user settings. This means we can used `ess-set-style` in hack-local-variables-hook.